euphonic_wrapper.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+import os
+from euphonic import Crystal, ForceConstants, QpointPhononModes, DebyeWaller, ureg
+from euphonic.util import mp_grid
+
+class EuphonicWrapper(object):
+    # This a wrapper around the Euphonic ForceConstants and QpointPhononModes classes to make it easier to access from Matlab
+    # It is meant to be used with a Matlab python_wrapper class and implements a horace_sqw function for use with Horace
+
+    defaults = {'force_constants': None, 'phonon_modes': None, 'debye_waller': None, 'debye_waller_grid': None, 
+                'temperature': 0.0 * ureg('K'), 'bose': True, 'negative_e': False, 'conversion_mat': None, 'chunk': 5000, 
+                'lim': np.inf, 'scattering_lengths': 'Sears1992', 'weights': None, 'asr': None, 'dipole': True, 
+                'eta_scale': 1.0, 'splitting': True, 'insert_gamma': False, 'reduce_qpts': True, 'use_c': False, 
+                'n_threads': 1, 'fall_back_on_python': True, 'convolution_function': 'gauss'}
+
+    def __init__(self, value=None, **kwargs):
+        for key, val in self.defaults.items():
+            setattr(self, key, kwargs.pop(key, self.defaults[key]))
+
+        if value is not None:
+            if hasattr(value, 'calculate_qpoint_phonon_modes'):
+                self.force_constants = value
+            elif hasattr(value, 'calculate_structure_factor'):
+                self.phonon_modes = value
+            elif isinstance(value, dict):
+                kwargs.update(value)
```

Comment:
Yeah, I was originally envisaging that this class would be the entry point from Matlab to all things Euphonic so it should include as properties all the main Euphonic classes, but since we've decided to have separate wrappers, it's now a bit redundant. I'll just remove this bit.